src/kvstore/kvstore_dist_sync_mpi.h

```diff
@@ -0,0 +1,175 @@
+#include <vector>
+#include <string>
+#include <utility>
+#include <functional>
+#include <algorithm>
+#include "./comm.h"
+#include "./kvstore_utils.h"
+
+#if MXNET_USE_MPI_DIST_KVSTORE
+#include "../mpi_collectives/include/mpi_collectives.h"
+
+namespace mxnet {
+namespace kvstore {
+
+/**
+ * \brief store data in local machine
+ */
+class KVStoreDistSyncMPI : public KVStore {
+ public:
+  KVStoreDistSyncMPI() : KVStore() {
+    int ret = MXMPIInit();
+    if (ret != 0) {
+      LOG(FATAL) << "kvstore with type [" << type_ << "] failed with mpi init";
+    }
+  }
+
+  virtual ~KVStoreDistSyncMPI() {
+  }
+
+  void Init(const std::vector<int>& keys,
+            const std::vector<NDArray>& values) override {
+    // Init does nothing in kvstore with type dist_sync_mpi
+  }
+
+  void Init(const std::vector<std::string>& str_keys,
+            const std::vector<NDArray>& values) override {
+    // Init does nothing in kvstore with type dist_sync_mpi
+  }
+
+  void Push(const std::vector<int>& keys,
+            const std::vector<NDArray>& values,
+            int priority) override {
+    LOG(WARNING) << "Not supported in KVStore with type " << type_ << ".";
+  }
+
+  void Pull(const std::vector<int>& keys,
+            const std::vector<NDArray*>& values,
+            int priority) override {
+    LOG(WARNING) << "Not supported in KVStore with type " << type_ << ".";
+  }
+
+  void PullRowSparse(const std::vector<int>& keys,
+                     const std::vector<std::pair<NDArray*, NDArray>>& val_rowids,
+                     int priority = 0) override {
+    LOG(WARNING) << "Not supported in KVStore with type " << type_ << ".";
+  }
+
+  void Push(const std::vector<std::string>& str_keys,
+            const std::vector<NDArray>& values,
+            int priority) override {
+    LOG(WARNING) << "Not supported in KVStore with type " << type_ << ".";
+  }
+
+  void Pull(const std::vector<std::string>& str_keys,
+            const std::vector<NDArray*>& values,
+            int priority) override {
+    LOG(WARNING) << "Not supported in KVStore with type " << type_ << ".";
+  }
+
+  void PullRowSparse(const std::vector<std::string>& str_keys,
+                     const std::vector<std::pair<NDArray*, NDArray>>& val_rowids,
+                     int priority = 0) override {
+    LOG(WARNING) << "Not supported in KVStore with type " << type_ << ".";
+  }
+
+  void SetGradientCompression(const std::vector<std::pair<std::string, std::string> >
+                              & kwargs) override {
+    LOG(WARNING) << "Not supported in KVStore with type " << type_ << ".";
+  }
+
+  void PushPull(const std::vector<int> &keys,
+                const std::vector<NDArray*> &in_values,
+                const std::vector<NDArray*> &out_values,
+                int priority) override {
+    int ret = MXMPIAllReduce(keys, in_values, out_values, priority);
+    if (ret != 0) {
+      LOG(WARNING) << "MXMPIAllReduce is not successful. ret: " << ret;
```

Comment:
Why is it a `LOG(WARNING)` instead of `LOG(FATAL)`?

Reply:
agree with you.